chain/network/src/network_protocol/proto_conv/account_key.rs

```diff
@@ -89,23 +80,13 @@ impl From<&SignedValidator> for proto::AccountKeySignedPayload {
     }
 }
 
-impl TryFrom<&proto::AccountKeySignedPayload> for SignedValidator {
-    type Error = ParseSignedValidatorError;
+impl TryFrom<&proto::AccountKeySignedPayload> for SignedAccountData {
+    type Error = ParseSignedAccountDataError;
     fn try_from(x: &proto::AccountKeySignedPayload) -> Result<Self, Self::Error> {
-        // We definitely should tolerate unknown fields, so that we can do
-        // backward compatible changes. We also need to limit the total
-        // size of the payload, to prevent large message attacks.
-        // TODO(CPR-74): is this the right place to do this check? Should we do the same while encoding?
-        // An alternative would be to do this check in the business logic of PeerManagerActor,
-        // probably together with signature validation. The amount of memory a node
-        // maintains per vaidator should be bounded.
-        if x.payload.len() > VALIDATOR_PAYLOAD_MAX_BYTES {
-            return Err(Self::Error::PayloadTooLarge(x.payload.len()));
-        }
```

Comment:
Duno, seems better to check the lengths during encode/decode, rather than during encode/in business logic.  Length limit seems like a property of the serialization format, it can be checked in isolation.

Reply:
Depends on how you look at that. It is responsibility of the account_data::Cache module to enforce a whole set of invariants to prevent abuse. This one would be an outlier property checked somewhere else. It is not like the size of a single message actually matters, but rather the total memory that a node uses to store broadcasted data.